File: backend/app/schemas/knowledge_article.py
from datetime import datetime

from pydantic import BaseModel


class KnowledgeArticleBase(BaseModel):
    title: str
    content: str
    category_id: int | None = None
    tags: str | None = None


class KnowledgeArticleCreate(KnowledgeArticleBase):
    pass


class KnowledgeArticleUpdate(BaseModel):
    title: str | None = None
    content: str | None = None
    category_id: int | None = None
    tags: str | None = None


class KnowledgeArticleResponse(KnowledgeArticleBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/knowledgeArticle.ts
export interface KnowledgeArticle {
  id: number
  title: string
  content: string
  category_id: number | null
  tags: string | null
  created_at: string
  updated_at: string
}

export interface KnowledgeArticleCreate {
  title: string
  content: string
  category_id?: number | null
  tags?: string | null
}

export interface KnowledgeArticleUpdate {
  title?: string
  content?: string
  category_id?: number | null
  tags?: string | null
}
